### ISSUE TITLE
[FormBuilderCheckboxList] Display FormBuilderFieldOption as Row instead of Column

### ISSUE THREAD

**Issue by awhitford** [danvick/flutter_form_builder#272]
When presenting something like a Checkbox List of Weekdays (Sun, Mon, Tue, Wed, Thu, Fri, Sat), displaying them vertically is taking a lot of space, where I would prefer to see them horizontally.

I think this is controlled by [this line](https://github.com/danvick/flutter_form_builder/blob/58d75452d718ca2a83b322a72a5b55ba39dcc9cb/lib/src/fields/form_builder_checkbox_list.dart#L160), but there presently is no control over it.

Could a configuration be added to `FormBuilderCheckboxList` to control whether the `FormBuilderFieldOption` are displayed as a _Column_ or a _Row_?

_(I haven't really tried this, so it may sound easier than it actually is.)_

**Comment by danvick** [comment]
Hi @awhitford,
I've been considering the use of [grouped_checkbox](https://pub.dev/packages/grouped_checkbox) package for this purpose but I think it may introduce one  or more breaking changes for current users.

I'm therefore planning to introduce it in v4.0.0 which is in the works.

**Comment by awhitford** [comment]
I've never used the `grouped_checkbox` package, but the flexibility that it seems to offer looks attractive.

If you introduced this as a new class like `FormBuilderCheckboxGroup`, and it behaved like `FormBuilderCheckboxList`, it could be (more or less) a drop-in replacement.  (Like how transitioning from `FormBuilderStepper` to `FormBuilderTouchSpin` was trivial.)

(For my particular scenario that triggered this issue, I ended up using a row of `CheckboxListTile` widgets.)

Another thing that caught my eye looking at [`grouped_checkbox`](https://github.com/zfnadia/grouped-checkbox) is those group borders.  I remember I used to use [`<fieldset>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) back with HTML.  Having a `field-group` widget might be nice.  (I know, separate issue.)
